newIDE/app/src/ResourcesList/LocalYarnBridge.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import optionalRequire from '../Utils/OptionalRequire.js';
+import { type ExternalEditorOpenOptions } from './ResourceExternalEditor.flow';
+import {
+  createOrUpdateResource,
+  getLocalResourceFullPath,
+} from './ResourceUtils.js';
+
+const electron = optionalRequire('electron');
+const path = optionalRequire('path');
+const ipcRenderer = electron ? electron.ipcRenderer : null;
+const gd = global.gd;
+
+/**
+ * Open YARN to Create/Edit Json Dialogue Tree resources.
+ */
+export const openYarn = ({
+  project,
+  resourcesLoader,
+  resourceNames,
+  onChangesSaved,
+  resourcePath,
+  extraOptions,
+}: ExternalEditorOpenOptions) => {
+  if (!electron || !ipcRenderer) return;
+  const projectPath = path.dirname(project.getProjectFile());
+  const initialResourcePath = getLocalResourceFullPath(
+    project,
+    resourceNames[0]
+  );
+
+  const externalEditorData = {
+    resourcePath: initialResourcePath,
+    // $FlowFixMe - TODO: There is an error here to be solved.
+    externalEditorData: extraOptions.externalEditorData,
+    projectPath,
+  };
+
+  ipcRenderer.removeAllListeners('yarn-changes-saved');
+  ipcRenderer.on(
+    'yarn-changes-saved',
+    (event, newFilePath, externalEditorData) => {
+      const resourceName = path.relative(projectPath, newFilePath);
+      createOrUpdateResource(project, new gd.JsonResource(), resourceName);
+
+      // $FlowFixMe - TODO: There is an error here to be solved.
+      onChangesSaved([{}], resourceName);
```

Comment:
Let's not introduce any FlowFixMe.
Here, you're calling the onChangesSaved, but the first argument is an array and you've added an empty object in the array.
Avoid empty objects, it does not make sense here, just pass an empty array `[]`.

Reply:
wasn't the metadata object used there? If so then, for now we can leave adding the name of the editor as metadata?

Reply:
ok I removed the fixflowme there :) we will introduce metadata for editor names at a later stage